homeassistant/components/tuya/config_flow.py

```diff
@@ -106,3 +144,237 @@ async def async_step_user(self, user_input=None):
+            dev_type = device.device_type()
+            if dev_type in TUYA_TYPE_CONFIG:
+                dev_id = f"{dev_type}-{device.object_id()}"
+                config_list[dev_id] = f"{device.name()} ({dev_type})"
+
+        return config_list
+
+    def _get_device(self, dev_id):
+        """Get specific device from tuya library."""
+        tuya = self.hass.data[DOMAIN][TUYA_DATA]
+        return tuya.get_device_by_id(dev_id)
+
+    def _save_config(self, data):
+        """Save the updated options."""
+        curr_conf = self.config_entry.options.copy()
+        curr_conf.update(data)
+        curr_conf.update(self._conf_devs_option)
+
+        return self.async_create_entry(title="", data=curr_conf)
+
+    async def _async_device_form(self, devs_id):
+        """Return configuration form for devices."""
+        count = 0
+        conf_devs_id = []
+        for dev_id in devs_id:
+            device_info = dev_id.split("-")
+            if count == 0:
+                device_type = device_info[0]
+                device_id = device_info[1]
+            elif device_type != device_info[0]:
+                self._form_error = ERROR_DEV_MULTI_TYPE
+                return await self.async_step_init()
+            conf_devs_id.append(device_info[1])
+            count += 1
+
+        device = self._get_device(device_id)
+        if not device:
+            self._form_error = ERROR_DEV_NOT_FOUND
+            return await self.async_step_init()
+
+        curr_conf = self._conf_devs_option.get(
+            device_id, self.config_entry.options.get(device_id, {})
+        )
+
+        config_schema = await self._get_device_schema(device_type, curr_conf, device)
+        if not config_schema:
+            self._form_error = ERROR_DEV_NOT_CONFIG
+            return await self.async_step_init()
+
+        self._conf_devs_id = conf_devs_id
+        device_name = "(multiple devices selected)" if count > 1 else device.name()
+
+        return self.async_show_form(
+            step_id="device",
+            data_schema=config_schema,
+            description_placeholders={
+                "device_type": device_type,
+                "device_name": device_name,
+            },
+        )
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            dev_ids = user_input.get(CONF_LIST_DEVICES)
+            if dev_ids:
+                return await self._async_device_form(dev_ids)
+
+            user_input.pop(CONF_LIST_DEVICES, [])
+            return self._save_config(data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_DISCOVERY_INTERVAL,
+                    default=self.config_entry.options.get(
+                        CONF_DISCOVERY_INTERVAL, DEFAULT_DISCOVERY_INTERVAL
+                    ),
+                ): vol.All(vol.Coerce(int), vol.Clamp(min=30, max=3600)),
```

Comment:
```suggestion
                ): vol.All(vol.Coerce(int), vol.Clamp(min=30, max=150)),
```

3600 seems high. Is there an actual use case for this?

Reply:
600 is too low. Current default value (without error) is 605. I agree that 3600 is very high, changed to 900